dll/win32/browseui/shellbrowser.cpp

```diff
@@ -3747,6 +3747,19 @@ LRESULT CShellBrowser::OnInitMenuPopup(UINT uMsg, WPARAM wParam, LPARAM lParam,
         menuIndex = 5;
     }
 
+    // Check the menu items for Explorer bar
+    BOOL bSearchBand = (IsEqualCLSID(CLSID_SH_SearchBand, fCurrentVertBar) ||
+                        IsEqualCLSID(CLSID_SearchBand, fCurrentVertBar) ||
+                        IsEqualCLSID(CLSID_IE_SearchBand, fCurrentVertBar) ||
+                        IsEqualCLSID(CLSID_FileSearchBand, fCurrentVertBar));
+    BOOL bHistory = IsEqualCLSID(CLSID_SH_HistBand, fCurrentVertBar);
+    BOOL bFavorites = IsEqualCLSID(CLSID_SH_FavBand, fCurrentVertBar);
+    BOOL bFolders = IsEqualCLSID(CLSID_ExplorerBand, fCurrentVertBar);
+    ::CheckMenuItem(theMenu, IDM_EXPLORERBAR_SEARCH, (bSearchBand ? MF_CHECKED : MF_UNCHECKED));
+    ::CheckMenuItem(theMenu, IDM_EXPLORERBAR_HISTORY, (bHistory ? MF_CHECKED : MF_UNCHECKED));
+    ::CheckMenuItem(theMenu, IDM_EXPLORERBAR_FAVORITES, (bFavorites ? MF_CHECKED : MF_UNCHECKED));
+    ::CheckMenuItem(theMenu, IDM_EXPLORERBAR_FOLDERS, (bFolders ? MF_CHECKED : MF_UNCHECKED));
```

Comment:
Why do you do it here, instead of inside the `else if (theMenu == SHGetMenuFromID(fCurrentMenuBar, FCIDM_MENU_VIEW))` or inside the `UpdateViewMenu()` function?
(I'm asking because these menu items seem to be inside that "explorer bar" submenu that is inside the "View" menu...)

Reply:
Fixed.

Reply:
:+1: